Firestore/core/test/firebase/firestore/util/filesystem_test.cc

```diff
@@ -66,21 +67,52 @@ TEST(FilesystemTest, Exists) {
   EXPECT_NOT_FOUND(IsDirectory(file));
 }
 
+#if defined(_WIN32)
+#define ASSERT_USEFUL_TEMP_DIR(dir)                          \
+  do {                                                       \
+    ASSERT_TRUE(absl::StartsWith(tmp.ToUtf8String(), "C:")); \
```

Comment:
In that case, you might consider just removing these altogether. They may be providing *some* value, but at least on some systems, they're incorrect.

Thinking about this a bit more, even on linux, although unusual, I don't think there's any reason TMPDIR needs to be an absolute path.

```
$ mktemp
/tmp/tmp.7QoTcJ8khC
$ TMPDIR=. mktemp
./tmp.pxADcfPMFH
```

Maybe the correct thing would be to check if the path indicated:
a) exists as a directory
b) is writable

Options (in my preferred order)
1) exists/writable
2) nothing
3) c: or / (i.e. as currently written).

I'm ok with any of the above.

Reply:
Setting `TMPDIR` to a relative path seems like a bad idea and while I can see how someone might do so, I see no great reason to accommodate our tests running in that configuration.

Meanwhile, someone could also set `TMPDIR` to a path that doesn't exist, is a readonly file, or is otherwise unsuitable for actual use, so none of the alternative tests you propose seem like they give any stronger guarantee. Boneheads gonna bone and all that.

The windows docs also specifically go out of their way to point out that the temporary directory does not have to exist.

Nevertheless, as a practical matter these tests all rely on `TempDir()` doing something useful so I've changed this to assert the result is nonempty and is a directory. I think we could also reasonably assert that the path is absolute, but I don't think that matters.